pkg/core/auth/authz/policy.go

```diff
@@ -0,0 +1,145 @@
+package authz
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/skygeario/skygear-server/pkg/core/auth"
+	"github.com/skygeario/skygear-server/pkg/server/skyerr"
+
+	"github.com/skygeario/skygear-server/pkg/gateway/model"
+)
+
+type PolicyProvider interface {
+	ProvideAuthzPolicy(*http.Request) Policy
```

Comment:
`ProvideAuthzPolicy` don't need to depends on the Request.

The policy should be same for all kind of request in a specific handler?

Reply:
This interface would be useful for restful api, and I imagine chat may be one of them.

Reply:
I think I will remove passing the request as param, and maybe we have a clearer picture on how to deal with it when doing chat.